int8 mode
---
int8 result
